src/app/static/src/app/utils.ts

```diff
@@ -190,6 +191,35 @@ export const flattenOptionsIdsByHierarchy = (filterOptions: NestedFilterOption[]
     return result;
 };
 
+export const flattenXAxisFilterOptionIds = (selections: BarchartSelections, filters: Filter[]) => {
```

Comment:
I guess it's a bit arbitrary, but this and `updateSelectionsAndXAxisOrder` could go in plot utils since they relate to barchart plots, just because there's already so much stuff in here. Or could even just be methods in the ModelOutput component (although that's also getting pretty bloated).